src/structure_managers/adaptor_neighbour_list.hh

```diff
@@ -940,15 +954,15 @@ namespace rascal {
 
       // outer mesh, including one layer of emtpy cells
       auto lmesh = std::fabs(mesh_min[i]) + max_coord + 3 * cutoff;
-      int n = std::ceil(lmesh / cutoff);
+      auto n = std::ceil(lmesh / cutoff);
```

Comment:
it is exactly what is intended here: all computation is done with doubles so there is no point to cast into an int to then cast it back to double the next line. So this modification keep the type consistent with double.

Reply:
Ok then, but you still have an implicit conversion when it's assigned into the array later; I would highly recommend making this conversion explicit (and using `double` instead of `auto` to clarify what is happening here).

Reply:
mesh_max is an Eigen::Matrix<double, traits::Dim, 1> so no implicit convertion